### ISSUE TITLE
Report Bugs: trigger rules are discarded in the latest implementation

### ISSUE THREAD

**Issue by zyf0726** [pietrobraione/jbse#32]
Hi, I I am facing an issue on executing triggers.
In *jbse.jvm.EngineParameters.java*, method *getTriggerRulesRepo* returns a **clone** of *this.triggerRulesRepo* but not *this.triggerRulesRepo* itself. 
```Java
public TriggerRulesRepo getTriggerRulesRepo() {
    final TriggerRulesRepo retVal = this.triggerRulesRepo.clone();
    if (getUseHashMapModel()) {
       	retVal.addExpandTo("java/util/HashMap", "(?!{°}*java/util/HashMap:initialMap{EOL}){°}*", "java/util/HashMap", new Signature("java/util/HashMap", "(Ljava/util/HashMap;)V", "initSymbolic"), "{$REF}");
    }
    return retVal;
}
```
This makes the operation in method *fillEngineParameters* of *jbse.apps.settings.SettingsReader.java* ineffective, since trigger rules are added to the temporary clone.
```Java
public void fillEngineParameters(EngineParameters params) {
    fillRulesTrigger(params.getTriggerRulesRepo());
    fillExpansionBackdoor(params.getExpansionBackdoor());
}
```

**Comment by pietrobraione** [comment]
Thank you for the report, and sorry for the late feedback.

**Comment by pietrobraione** [comment]
Closed by pull request #36.
